File: macros/type_test.cpp
#include <TFile.h>
#include <TTree.h>
#include <typeinfo>

void type_test() {
  TFile* mfile = TFile::Open("new100.root");

  TTree* mtree = (TTree*)mfile->Get("h1");

  Float_t mvalue;
  mtree->SetBranchAddress("X_vtx", &mvalue);

  mtree->GetEntry(0);
  
  printf("Type name of the first value in branch X_vtx: %f %s\n", mvalue, typeid(mvalue).name());

  mfile->Close();

  TFile* file = TFile::Open("../batch/data/AnaBarMC_Gen_7001.root");

  TTree* tree = (TTree*)file->Get("h1");

  Float_t value;
  tree->SetBranchAddress("X_vtx", &value);

  tree->GetEntry(0);
  
  printf("Type name of the first value in branch X_vtx: %f %s\n", value, typeid(value).name());

  file->Close();
}

File: macros/type_test.py
import uproot
import particle

# Open the G4SBS root file and navigate to the tree containing the event data
myfile = uproot.open("new100.root")
mytree = myfile["h1"]

bfile = uproot.open("../batch/data/AnaBarMC_Gen_7001.root")
btree = bfile["h1"]

print("\nnew100:")
print(mytree.typenames())

print("\nAnaBarMC_Gen_7001:")
print(btree.typenames())

for branch in btree:
	print(branch.name)
	print(branch.array()[0])

'''
for branch in mytree:
	print()
	print()
	print(branch.name)
	print(branch.values())
	myx = mytree[branch.name].array()
	print("mine:")
	print(type(myx))
	print(myx[0])
	print(type(myx[0]))
	
	print()
	
	bx = btree[branch.name].array()
	print("his:")
	print(type(bx))
	print(bx[0])
	print(type(bx[0]))
'''
